src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1101,8 +1146,40 @@ FixedwingPositionControl::handle_setpoint_type(const uint8_t setpoint_type, cons
 	return position_sp_type;
 }
 
-void
-FixedwingPositionControl::control_auto_position(const hrt_abstime &now, const float dt, const Vector2d &curr_pos,
+Vector2f FixedwingPositionControl::navigateWaypoints(const Vector2f &waypoint_A, const Vector2f &waypoint_B,
+		const Vector2f &vehicle_pos)
+{
+	Vector2f unit_path_tangent;
+	// similar to logic found in ECL_L1_Pos_Controller method of same name
+	// BUT no arbitrary max approach angle, approach entirely determined by generated
+	// bearing vectors
+
+	Vector2f vector_A_to_B = waypoint_A - waypoint_B;
+	Vector2f vector_A_to_vehicle = waypoint_A - vehicle_pos;
+
+	if (vector_A_to_B.norm() < FLT_EPSILON) {
+		// the waypoints are on top of each other and should be considered as a
+		// single waypoint, fly directly to it
+		unit_path_tangent(0) = NAN;
+		unit_path_tangent(1) = NAN;
+
+	} else if (vector_A_to_B.dot(vector_A_to_vehicle) < 0.0f) {
+		// we are in front of waypoint A, fly directly to it until the bearing generated
```

Comment:
This comment is implying that you still use the logic that was in NPFG navigate function where the bearing generated from guideToPath() is shallower than guideToPoint()

https://github.com/PX4/PX4-Autopilot/blob/8de2c80b34598e7eb908cecf370bae7b33dc3b24/src/lib/npfg/npfg.cpp#L541-L565

Also .. would be nice to keep this logic! maybe pathhandler then should have access to the guidance controller?

Reply:
@tstastny Ah, good that you spotted this. I was struggling to use this logic as is since it was relying on an internal state `bearing_vec_`. Would you have any suggestions on how we can remove this?

Reply:
@Jaeyoung-Lim ideally there would be no internal state `bearing_vec_` for NPFG. I only implemented the internal state such that similar "getter" functions could be called as L1 at each point in the fw pos ctrl loop. However, the bearing vector is still basically just used as output. It is not changed unless a `guideToXXX()` method is called. The original intent was that the "navigate" methods are *not part of the NPFG library, and rather part of a yet to be created (until you may have done so in this PR!) "path handler". Then the `guideToXXX()` methods could be exposed as public in NPFG, and would be the only interface, probably with param[out]'s for each respective output (bearing vec, airspeed setpoint, acceleration command, etc). Then we can get rid of these internal states.